Shopify/shopify-cli
# Update shopify-extensions to v0.3.0 and fix issues with template choosing and missing dependencies

### WHY are these changes introduced?
We need to consume new changes from the shopify-extensions Go binary in order to support localization in Checkout UI extensions.

### WHAT is this pull request doing?
Bump the version of shopify-extensions to v0.3.0 and fix issues with template choosing and missing dependencies.

https://user-images.githubusercontent.com/29458473/162487247-ad5c30bc-6f15-4c71-8df0-4d2fe9ae1d87.mov

### How to test your changes?
1. Pull this PR and run `rake extensions:install` in order to install the GO binary. For more information on testing the integration please see the [`shopify-cli-extensions` README](https://github.com/Shopify/shopify-cli-extensions#testing-the-integration-with-the-shopify-cli)
2. Create an alias in your terminal setting to point to the `bin/shopify` file of this repo. For example
    ```
    alias sd='/Users/trishta/src/github.com/Shopify/shopify-cli/bin/shopify'
    ```
4. In a new folder run `sd config feature extension_server_beta --enable`
5. Run `sd extension create`
6. When prompted select the app with `checkout_ui_extension` enabled
7. When prompted select "Checkout Extension" for the extension type
8. When prompted select "typescript-react"
9. When prompted enter any name for the extension
10. Navigate to the extension folder
11. Run `sd extensions serve`
12. Ensure that you see the server runs and you see a deep link to go to a checkout with the local extension loaded

### Post-release steps



### Update checklist
